File: app/src/main/java/com/widyatama/nurseassistant/features/healingPlan/HealingPlanPresenter.java
package com.widyatama.nurseassistant.features.healingPlan;

import com.widyatama.core.base.BasePresenter;
import com.widyatama.core.util.SchedulerProviderUtil;
import com.widyatama.nurseassistant.data.local.LocalStorageHelper;

import io.reactivex.disposables.CompositeDisposable;

public class HealingPlanPresenter<V extends HealingPlanViewContracts> extends BasePresenter<V>
    implements HealingPlanPresenterContracts<V> {

    LocalStorageHelper localStorage;
    SchedulerProviderUtil schedulerProvider;
    CompositeDisposable compositeDisposable;

    public HealingPlanPresenter(LocalStorageHelper localStorage, SchedulerProviderUtil schedulerProvider,
                                CompositeDisposable compositeDisposable){
        super(schedulerProvider, compositeDisposable);

        this.localStorage = localStorage;
        this.schedulerProvider = schedulerProvider;
        this.compositeDisposable = compositeDisposable;
    }

    @Override
    public void getHealing(int limit) {
        compositeDisposable.add(localStorage.getSampleDatabase().healingPlan().getAll()
            .compose(schedulerProvider.ioToMainFlowableScheduler())
            .subscribe(healingPlans -> {
                view.showHealing(healingPlans);
            }, throwable -> System.out.println("Error get healing")));
    }
}

File: app/src/main/java/com/widyatama/nurseassistant/features/healingPlan/HealingPlanPresenter.kt
package com.widyatama.nurseassistant.features.healingPlan

import com.widyatama.core.base.BasePresenter
import com.widyatama.core.util.SchedulerProviderUtil
import com.widyatama.nurseassistant.R
import com.widyatama.nurseassistant.data.local.LocalStorageHelper
import com.widyatama.nurseassistant.data.model.HealingPlan
import io.reactivex.disposables.CompositeDisposable

class HealingPlanPresenter<V: HealingPlanViewContracts>
constructor(private var localStorage: LocalStorageHelper, schedulerProvider: SchedulerProviderUtil,
            compositeDisposable: CompositeDisposable) : BasePresenter<V>(
        schedulerProvider, compositeDisposable), HealingPlanPresenterContracts<V> {

    override fun getHealing(limit: Int) {
        compositeDisposable.add(localStorage.sampleDatabase.healingPlan().getAll()
                .compose(schedulerProvider.ioToMainFlowableScheduler())
                .subscribe({
                    view?.showHealing(it)
                }, {
                    print("error get Healing")
                }))
    }
}